### PR TITLE
Avular Backport

### PR DESCRIPTION
**Pull request by ar13pit**

### CODE REVIEW

File: ci/mirror-upstream.bash [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+# Script to mirror a branch from a different remote
+
+# Stop on errors
+set -o errexit
+
+# Execute script only in a CI environment
+if [ "$CI" != "true" ]
+then
+    echo -e "\e[35m\e[1m Error!\e[0m Trying to execute a CI script in a non-CI environment. Exiting script."
```
Comment (MatthijsBurgh) [inline ci/mirror-upstream.bash:11]: Remove space

File: ci/mirror-upstream.bash [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+git fetch origin
+git checkout "$LOCAL_REMOTE_BRANCH" || { echo "Could not checkout local remote branch"; exit 1; }
+git merge fork/"$UPSTREAM_REMOTE_BRANCH" --ff-only || { echo "Could not sync with upstream remote branch"; exit 1; }
+
+git remote show origin
+git remote set-url --push origin "$LOCAL_REMOTE_URL"
+git push origin "$LOCAL_REMOTE_BRANCH"
+
+echo
+echo -e "\e[35m\e[1m Synced local branch '$LOCAL_REMOTE_BRANCH' with upstream remote branch '$UPSTREAM_REMOTE_BRANCH'"
```
Comment (MatthijsBurgh) [inline ci/mirror-upstream.bash:75]: Remove spaces

File: ci/test-package.sh [hunk 3]
```diff
@@ -10,7 +10,7 @@ set -o errexit
 # Execute script only in a CI environment
 if [ "$CI" != "true" ]
 then
-    echo -e "\e[35m\e[1mError!\e[0mTrying to execute a CI script in a non-CI environment. Exiting script."
+    echo -e "\e[35m\e[1m Error!\e[0m Trying to execute a CI script in a non-CI environment. Exiting script."
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:13]: Remove spaces

File: ci/test-package.sh [hunk 4]
```diff
@@ -23,27 +23,58 @@ do
 
         * )
             # unknown option
-            echo -e "\e[35m\e[1mUnknown input argument '$i'. Check CI .yml file\e[0m"
+            echo -e "\e[35m\e[1m Unknown input argument '$i'. Check CI .yml file \e[0m"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:26]: Remove spaces

File: ci/test-package.sh [hunk 5]
```diff
@@ -23,27 +23,58 @@ do
...
             exit 1 ;;
     esac
     shift
 done
 
-echo -e "\e[35m\e[1mPACKAGE     = ${PACKAGE}\e[0m"
+echo -e "\e[35m\e[1m PACKAGE     = ${PACKAGE} \e[0m"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:32]: Remove spaces

File: ci/test-package.sh [hunk 6]
```diff
@@ -23,27 +23,58 @@ do
...
 
 # If packages is non-zero, this is a multi-package repo. In multi-package repo, check if this package needs CI.
 # If a single-package repo, CI is always needed.
 # shellcheck disable=SC2153
 if [ -n "$PACKAGES" ] && ! echo "$PACKAGES" | grep -sqw "$PACKAGE"
 then
-    echo -e "\e[35m\e[1mNo changes in this package, so no need to run CI\e[0m"
+    echo -e "\e[35m\e[1m No changes in this package, so no need to run CI \e[0m"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:39]: Remove spaces

File: ci/test-package.sh [hunk 7]
```diff
@@ -23,27 +23,58 @@ do
...
+docker exec -i tue-env bash <<EOF
+#!/bin/bash
+
+set -e -o pipefail
+
+source ~/.bashrc
+
+if [ "\${TUE_ROS_VERSION}" == "1" ]; then
+    cd "\$TUE_SYSTEM_DIR"/src/"\$PACKAGE"
+    echo -e "\e[35m\e[1m catkin test --this --no-status --no-deps \e[0m"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:53]: Remove spaces

File: ci/test-package.sh [hunk 8]
```diff
@@ -23,27 +23,58 @@ do
...
+    catkin test --this --no-status --no-deps
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:54]: Missing `catkin_enable_testing`

File: ci/test-package.sh [hunk 9]
```diff
@@ -23,27 +23,58 @@ do
...
+elif [ "\${TUE_ROS_VERSION}" == "2" ]; then
+    cd "\$TUE_SYSTEM_DIR"
+    echo -e "\e[35m\e[1m colcon test --merge-install --event-handlers status- console_direct+ --return-code-on-test-failure --packages-select \$PACKAGE \e[0m"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:57]: Do something here with the testing too

File: ci/test-package.sh [hunk 10]
```diff
@@ -23,27 +23,58 @@ do
...
+#!/bin/bash
+
+set -e -o pipefail
+
+source ~/.bashrc
+
+if [ "\${TUE_ROS_VERSION}" == "1" ]; then
+    cd "\$TUE_SYSTEM_DIR"
+    echo -e "\e[35m\e[1m catkin_test_results build/\$PACKAGE \e[0m"
+    catkin_test_results build/"\$PACKAGE"
```
Comment (MatthijsBurgh) [inline ci/test-package.sh:74]: This is include in `catkin test`

File: dockerfiles/tue-env.Dockerfile [hunk 11]
```diff
@@ -28,11 +32,15 @@ SHELL ["/bin/bash", "-c"]
...
-    # Add amigo user
+    echo "resolvconf resolvconf/linkify-resolvconf boolean false" | debconf-set-selections && \
+    apt-get install -qq --assume-yes --no-install-recommends apt-transport-https apt-utils bash-completion ca-certificates curl dbus debconf-utils dialog git keyboard-configuration lsb-release iproute2 iputils-ping net-tools openssh-client psmisc resolvconf sudo tzdata wget > /dev/null && \
+    # Add defined user
     adduser -u 1000 --disabled-password --gecos "" $USER && \
     usermod -aG sudo $USER && \
     usermod -aG adm $USER && \
+    usermod -aG dialout $USER && \
+    usermod -aG video $USER && \
+    usermod -aG audio $USER && \
```
Comment (MatthijsBurgh) [inline dockerfiles/tue-env.Dockerfile:43]: Why?

File: dockerfiles/tue-env.Dockerfile [hunk 12]
```diff
@@ -28,11 +32,15 @@ SHELL ["/bin/bash", "-c"]
 # Install commands used in our scripts and standard present on a clean ubuntu
 # installation and setup a user with sudo priviledges
 RUN apt-get update -qq && \
-    apt-get install -qq --assume-yes --no-install-recommends apt-transport-https apt-utils ca-certificates dbus dialog git keyboard-configuration lsb-release openssh-client sudo tzdata wget > /dev/null && \
-    # Add amigo user
+    echo "resolvconf resolvconf/linkify-resolvconf boolean false" | debconf-set-selections && \
+    apt-get install -qq --assume-yes --no-install-recommends apt-transport-https apt-utils bash-completion ca-certificates curl dbus debconf-utils dialog git keyboard-configuration lsb-release iproute2 iputils-ping net-tools openssh-client psmisc resolvconf sudo tzdata wget > /dev/null && \
```
Comment (MatthijsBurgh) [inline dockerfiles/tue-env.Dockerfile:36]: Please explain the new packages

Comment (MatthijsBurgh) [inline dockerfiles/tue-env.Dockerfile:36]: I don't like the `-qq`. As when `apt-get` fails with double quite. The logs are so short, it is almost impossible to find the problem.

File: dockerfiles/tue-env.Dockerfile [hunk 13]
```diff
@@ -55,14 +63,13 @@ RUN --mount=type=ssh,uid=1000 sed -e s/return//g -i ~/.bashrc && \
...
     # Run the standard installation script
-    source bootstrap.bash && \
+    source bootstrap.bash --ros-version="$ROS_VERSION" --ros-distro="$ROS_DISTRO" && \
     # Make tue-env to be available to the environment
     source ~/.bashrc && \
-    # Set all git repositories to use HTTPS urls (Needed for local image builds)
-    tue-env config ros-"$TUE_ROS_DISTRO" git-use-https && \
     # Install target ros
-    tue-get install ros --test-depend && \
+    tue-get install ros${ROS_VERSION} --test-depend --branch="$BRANCH" && \
```
Comment (MatthijsBurgh) [inline dockerfiles/tue-env.Dockerfile:72]: Conflicting with our targets

File: dockerfiles/tue-env.Dockerfile [hunk 14]
```diff
@@ -72,4 +79,6 @@ RUN --mount=type=ssh,uid=1000 sed -e s/return//g -i ~/.bashrc && \
     # Check git remote origin
     git -C ~/.tue remote -v && \
     # Show the branches of tue-env
-    git -C ~/.tue branch
+    git -C ~/.tue branch && \
+    # Remove docker-clean. APT will be able to autocomplete packages now
+    sudo rm /etc/apt/apt.conf.d/docker-clean
```
Comment (MatthijsBurgh) [inline dockerfiles/tue-env.Dockerfile:84]: Elaborate please

File: setup/tue-env-config.bash [hunk 15]
```diff
@@ -108,6 +108,15 @@ function tue-env-not-install-doc-depend
     echo -e "[tue-env](config) Environment '$env' set to not install doc dependencies"
 }
 
+function tue-env-set
```
Comment (MatthijsBurgh) [inline setup/tue-env-config.bash:111]: Is this documented in the README?

**Review by MatthijsBurgh — CHANGES_REQUESTED** [review]
Still no to review some files.

**Comment by ar13pit** [comment]
@MatthijsBurgh why did you review this and fill it with comments when this was still work in progress? :angry:

**Comment by ar13pit** [comment]
I'm closing this PR because cleaning it up with all these review comments is a bigger headache

**Pull request closed without merging**